### ISSUE TITLE
[1.10] Minecolonies crash

### ISSUE THREAD

**Issue by hron84** [ldtteam/minecolonies#3205]
### Minecolonies version

Version: 0.9.7587 (from Curse)

### Expected behavior

In singleplayer, when I first start the world, it works well. However, after save the world and try to reload (for example to fix a chunk not loading issue) this crash happens.

```
// Ooh. Shiny.

Time: 12/29/18 11:13 AM
Description: Exception in server tick loop

java.lang.NullPointerException: Exception in server tick loop
	at com.minecolonies.api.compatibility.CompatabilityManager.discover(CompatabilityManager.java:67)
	at com.minecolonies.coremod.colony.ColonyManager.readFromNBT(ColonyManager.java:871)
	at com.minecolonies.coremod.colony.ColonyManager.onWorldLoad(ColonyManager.java:723)
	at com.minecolonies.coremod.event.EventHandler.onWorldLoad(EventHandler.java:398)
	at net.minecraftforge.fml.common.eventhandler.ASMEventHandler_527_EventHandler_onWorldLoad_Load.invoke(.dynamic)
	at net.minecraftforge.fml.common.eventhandler.ASMEventHandler.invoke(ASMEventHandler.java:90)
	at net.minecraftforge.fml.common.eventhandler.EventBus.post(EventBus.java:185)
	at net.minecraft.server.integrated.IntegratedServer.func_71247_a(IntegratedServer.java:96)
	at net.minecraft.server.integrated.IntegratedServer.func_71197_b(IntegratedServer.java:120)
	at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:431)
	at java.lang.Thread.run(Unknown Source)

-- System Details --
Details:
	Minecraft Version: 1.10.2
	Operating System: Windows 10 (amd64) version 10.0
	Java Version: 1.8.0_191, Oracle Corporation
	Java VM Version: Java HotSpot(TM) 64-Bit Server VM (mixed mode), Oracle Corporation
	Memory: 2669968488 bytes (2546 MB) / 4752670720 bytes (4532 MB) up to 5726797824 bytes (5461 MB)
	JVM Flags: 4 total; -Xmx6144m -Xms256m -XX:PermSize=256m -XX:HeapDumpPath=MojangTricksIntelDriversForPerformance_javaw.exe_minecraft.exe.heapdump
	IntCache: cache: 0, tcache: 0, allocated: 12, tallocated: 94
	FML: MCP 9.32 Powered by Forge 12.18.3.2511 76 mods loaded, 76 mods active
	States: 'U' = Unloaded 'L' = Loaded 'C' = Constructed 'H' = Pre-initialized 'I' = Initialized 'J' = Post-initialized 'A' = Available 'D' = Disabled 'E' = Errored
	UCHIJAAAAAAA	mcp{9.19} [Minecraft Coder Pack] (minecraft.jar) 
	UCHIJAAAAAAA	FML{8.0.99.99} [Forge Mod Loader] (forge-1.10.2-12.18.3.2511.jar) 
	UCHIJAAAAAAA	Forge{12.18.3.2511} [Minecraft Forge] (forge-1.10.2-12.18.3.2511.jar) 
	UCHIJAAAAAAA	appliedenergistics2-core{rv4-stable-1} [Applied Energistics 2 Core] (minecraft.jar) 
	UCHIJAAAAAAA	Aroma1997Core{${version}} [Aroma1997Core] (Aroma1997Core-1.9.4-1.1.1.2.jar) 
	UCHIJAAAAAAA	<CoFH ASM>{000} [CoFH ASM] (minecraft.jar) 
	UCHIJAAAAAAA	ccl-entityhook{1.0} [ccl-entityhook] (CodeChickenLib-1.10.2-2.5.9.320-universal.jar) 
	UCHIJAAAAAAA	cofhcore{4.1.12} [CoFH Core] (CoFHCore-1.10.2-4.1.12.17-universal.jar) 
	UCHIJAAAAAAA	actuallyadditions{1.10.2-r105} [Actually Additions] (ActuallyAdditions-1.10.2-r105.jar) 
	UCHIJAAAAAAA	embassy{1.1.2} [Embassy] (Embassy 1.10.2-1.1.2.117.jar) 
	UCHIJAAAAAAA	actualmusic{2.0.4} [ActualMusic] (ActualMusic 2.0.4.133.jar) 
	UCHIJAAAAAAA	Baubles{1.3.13} [Baubles] (Baubles-1.10.2-1.3.13.jar) 
	UCHIJAAAAAAA	JEI{3.14.7.420} [Just Enough Items] (jei_1.10.2-3.14.7.420.jar) 
	UCHIJAAAAAAA	extrautils2{1.0} [Extra Utilities 2] (extrautils2-1.10.2-1.7.5.jar) 
	UCHIJAAAAAAA	flyringbaublemod{0.2.4_1.10.2-d71eb9e} [AngelRing 2 Bauble] (angelRingToBauble-1.10.2-0.2.4.19+d71eb9e.jar) 
	UCHIJAAAAAAA	AppleCore{3.1.1} [AppleCore] (AppleCore-mc1.10.2-3.1.1.jar) 
	UCHIJAAAAAAA	appleskin{1.0.9} [AppleSkin] (AppleSkin-mc1.10.2-1.0.9.jar) 
	UCHIJAAAAAAA	appliedenergistics2{rv4-stable-1} [Applied Energistics 2] (appliedenergistics2-rv4-stable-1.jar) 
	UCHIJAAAAAAA	aquaculture{1.4.1} [Aquaculture] (Aquaculture-1.10.2-1.4.1.4.jar) 
	UCHIJAAAAAAA	Aroma1997CoreHelper{1.1.1.2} [Aroma1997Core|Helper] (Aroma1997Core-1.9.4-1.1.1.2.jar) 
	UCHIJAAAAAAA	AromaBackup{1.0.1.1} [AromaBackup] (AromaBackup-1.9.4-1.0.1.1.jar) 
	UCHIJAAAAAAA	AromaBackupRecovery{1.0} [AromaBackup Recovery] (AromaBackup-1.9.4-1.0.1.1.jar) 
	UCHIJAAAAAAA	astralsorcery{1.4.5} [Astral Sorcery] (astralsorcery-1.4.5.jar) 
	UCHIJAAAAAAA	betterbuilderswands{0.11.1} [Better Builder's Wands] (BetterBuildersWands-1.10.2-0.11.1.220+f8232fe.jar) 
	UCHIJAAAAAAA	BiblioCraft{2.2.5} [BiblioCraft] (BiblioCraft[v2.2.5][MC1.10.2].jar) 
	UCHIJAAAAAAA	CodeChickenLib{2.5.9.320} [CodeChicken Lib] (CodeChickenLib-1.10.2-2.5.9.320-universal.jar) 
	UCHIJAAAAAAA	CodeChickenCore{2.4.1.102} [CodeChicken Core] (CodeChickenCore-1.10.2-2.4.1.102-universal.jar) 
	UCHIJAAAAAAA	ChickenChunks{2.2.0.52} [ChickenChunks] (ChickenChunks-1.10.2-2.2.0.52-universal.jar) 
	UCHIJAAAAAAA	ctm{MC1.10.2-0.3.2.35} [CTM] (CTM-MC1.10.2-0.3.2.35.jar) 
	UCHIJAAAAAAA	chisel{MC1.10.2-0.2.1.47} [Chisel] (Chisel-MC1.10.2-0.2.1.47.jar) 
	UCHIJAAAAAAA	controlling{1.0.5} [Controlling] (Controlling-1.0.5.jar) 
	UCHIJAAAAAAA	cookingforblockheads{4.2.26} [Cooking for Blockheads] (CookingForBlockheads_1.10.2-4.2.26.jar) 
	UCHIJAAAAAAA	crafttweakerjei{1.0.1} [CraftTweaker JEI Support] (CraftTweaker-1.10.2-3.0.26.jar) 
	UCHIJAAAAAAA	MineTweaker3{3.0.26} [MineTweaker 3] (CraftTweaker-1.10.2-3.0.26.jar) 
	UCHIJAAAAAAA	ctgui{1.0.0} [CT-GUI] (CraftTweaker-1.10.2-3.0.26.jar) 
	UCHIJAAAAAAA	customstartinggear{1.0.0} [Custom Starting Gear] (CustomStartingGear-1.10.2-1.0.0.7-universal.jar) 
	UCHIJAAAAAAA	EnderStorage{2.2.1.106} [EnderStorage] (EnderStorage-1.10.2-2.2.1.106-universal.jar) 
	UCHIJAAAAAAA	fasterladderclimbing{0.1-122} [Faster Ladder Climbing] (FasterLadderClimbing-1.9.4-0.1-122.jar) 
	UCHIJAAAAAAA	fastleafdecay{v11} [Fast Leaf Decay] (FastLeafDecay-v11.jar) 
	UCHIJAAAAAAA	harvestfestival{0.5.27} [Harvest Festival] (Harvest-Festival-1.10.2-0.5.27.jar) 
	UCHIJAAAAAAA	harvestcraft{1.9.4-1.10.2g} [Pam's HarvestCraft] (Pam's HarvestCraft 1.9.4-1.10.2h.jar) 
	UCHIJAAAAAAA	hfaddon{0.5.5 Alpha} [HF Addon] (fhaddon-0.5.5.jar) 
	UCHIJAAAAAAA	com.ch1a.flatworld{1.4.2} [flatworld] (FlatWorld1.4.2-1.9-1.10.jar) 
	UCHIJAAAAAAA	Waila{1.8.17-B31_1.10.2} [Waila] (Hwyla-1.8.17-B31_1.10.2.jar) 
	UCHIJAAAAAAA	mcmultipart{1.4.0} [MCMultiPart] (MCMultiPart-1.4.0-universal.jar) 
	UCHIJAAAAAAA	ftbl{0.0.0} [FTBLib] (FTBLib-1.1x-3.6.5.jar) 
	UCHIJAAAAAAA	gravestone{1.5.13} [Gravestone Mod] (gravestone-1.5.13.jar) 
	UCHIJAAAAAAA	inventorytweaks{1.61-58-a1fd884} [Inventory Tweaks] (InventoryTweaks-1.61-58.jar) 
	UCHIJAAAAAAA	ironbackpacks{1.10.2-2.2.33} [Iron Backpacks] (IronBackpacks-1.10.2-2.2.33.jar) 
	UCHIJAAAAAAA	jehc{1.2.2} [Just Enough HarvestCraft] (jehc-1.10.2-1.2.2.jar) 
	UCHIJAAAAAAA	journeymap{1.10.2-5.5.2} [JourneyMap] (journeymap-1.10.2-5.5.2.jar) 
	UCHIJAAAAAAA	jeresources{0.5.8.98} [Just Enough Resources] (JustEnoughResources-1.10.2-0.5.8.98.jar) 
	UCHIJAAAAAAA	levels{3.0.2} [Levels 3] (Levels 3-1.10.2-3.0.2-beta.jar) 
	UCHIJAAAAAAA	llor{1.1.1-mc[1.9-1.11]} [Light Level Overlay Reloaded] (LLOverlayReloaded-1.1.1-mc[1.9-1.11].jar) 
	UCHIJAAAAAAA	mantle{1.10.2-1.1.5.205} [Mantle] (Mantle-1.10.2-1.1.5.jar) 
	UCHIJAAAAAAA	minecolonies{@VERSION@} [MineColonies] (minecolonies-universal-1.10.2-0.9.7587.jar) 
	UCHIJAAAAAAA	mtrm{1.2.1.26} [MineTweakerRecipeMaker] (MineTweakerRecipeMaker-1.10.2-1.2.1.26.jar) 
	UCHIJAAAAAAA	mousetweaks{2.8} [Mouse Tweaks] (MouseTweaks-2.8-mc1.10.2.jar) 
	UCHIJAAAAAAA	oreexcavation{1.2.131} [OreExcavation] (OreExcavation-1.2.131.jar) 
	UCHIJAAAAAAA	tconstruct{1.10.2-2.6.5.10} [Tinkers' Construct] (TConstruct-1.10.2-2.6.5.jar) 
	UCHIJAAAAAAA	p455w0rdslib{1.0.35} [p455w0rd's Library] (p455w0rdslib-1.10.2-1.0.35.jar) 
	UCHIJAAAAAAA	refinedstorage{1.2.26} [Refined Storage] (refinedstorage-1.2.26.jar) 
	UCHIJAAAAAAA	stg{1.10.2-1.0.3} [SwingThroughGrass] (stg-1.10.2-1.0.3.jar) 
	UCHIJAAAAAAA	lteleporters{1.1.3} [Lemon's Simple Teleporters] (teleporters-1.1.3.jar) 
	UCHIJAAAAAAA	thaumcraft{6.0.BETA8} [Thaumcraft] (Thaumcraft-1.10.2-6.0.BETA8.jar) 
	UCHIJAAAAAAA	tcinventoryscan{2.0.7} [Thaumcrafft Inventory Scanning] (ThaumcraftInventoryScanning_1.10.2-2.0.7.jar) 
	UCHIJAAAAAAA	ThaumicJEI{1.1.2} [ThaumicJEI] (ThaumicJEI-1.10.2-1.1.3-2.jar) 
	UCHIJAAAAAAA	thermalfoundation{2.1.5} [Thermal Foundation] (ThermalFoundation-1.10.2-2.1.5.12-universal.jar) 
	UCHIJAAAAAAA	thermalexpansion{5.1.10} [Thermal Expansion] (ThermalExpansion-1.10.2-5.1.10.28-universal.jar) 
	UCHIJAAAAAAA	thermaldynamics{2.0.11} [Thermal Dynamics] (ThermalDynamics-1.10.2-2.0.11.19-universal.jar) 
	UCHIJAAAAAAA	wanionlib{1.10.2-1.3} [WanionLib] (WanionLib-1.10.2-1.3.jar) 
	UCHIJAAAAAAA	icse{1.1.0.0} [I Can See Everything] (Wawla-1.10.2-2.3.2.215.jar) 
	UCHIJAAAAAAA	wawla{2.3.2.215} [What Are We Looking At] (Wawla-1.10.2-2.3.2.215.jar) 
	UCHIJAAAAAAA	wcg{1.0.0} [Wireless Crafting Grid] (WirelessCraftingGrid-1.10.2-1.0.0.jar) 
	UCHIJAAAAAAA	yabba{1.10.2-0.5.1} [YABBA] (YABBA-1.10.2-0.5.1.jar) 
	UCHIJAAAAAAA	unidict{1.10.2-2.9b} [UniDict] (UniDict-1.10.2-2.9b.jar) 
 ```

### Actual behaviour

This should not happen

### Steps to reproduce the problem

1. Create a world
2. Save and exit to main menu
3. Try to re-load the world
4. ... Profit

PS: Could be an issue if I have a space in the world's name? I'll try to rename to see what happens, but I hope it's not an issue. However, when reproduce, put a space into the name of the world.

**Comment by marchermans** [comment]
Sorry 1.10 is more then 2 Versions behind the current release and as such is no longer supported. We tried to support 1.10 as Long as we could. But this is simply not possible anymore. We do not have the man power to Keep that Version running and supported in the way we feel our community deserves. Could you update your instance to 1.12?

**Comment by hron84** [comment]
@OrionDevelopment Sadly, this is a modpack that I tried to spice up with MineColonies (it's Farming Harvest, similar to Farming Valley, but a bit different tech and magic setup) so I think it would not be possible to upgrade it to 1.12. 

I took a look into this issue, and as I think something registers a null into OreDictionary, so probably this issue is not on your side. However, consider a null-check in discover codes in the CompatibilityManager as it seems like there are buggy mods in the wild.
